en/docs/identity-providers/exchange-an-access-token-from-an-external-identity-provider.md

```diff
@@ -0,0 +1,43 @@
+# Exchange an Access Token from an External Identity Provider (IdP)
+
+After you [Register an External Identity Provider]({{base_path}}/identity-providers/register-an-external-identity-provider) you can exchange access tokens from that IdP for Choreo access tokens and use those to invoke APIs. 
+
+Follow this procedure to exchange an access token from an external IdP for a Choreo token within your organization.
+
+1. Sign in to the Choreo Developer Portal at [https://devportal.choreo.dev/](https://devportal.choreo.dev/).
+
+2. Click **Applications** tab.
+
+3. Create a new application or use an existing application.
+
+4. Select **OAuth2 Tokens** under the **Production Keys** on the left bar of the application.
+
+5. If the Consumer Key and the Consumer Secret have not been generated before, the following page will be displayed. Click on **Generate Keys** to generate the Consumer Key and the Consumer Secret.
+
+    ![OAuth2 Tokens](../assets/img/identity-providers/oauth2-tokens.png){.cInlineImage-full}
+
+    !!! tip
+        The **User Keys** tab will be disabled if you did not generate the keys or if you have generated the keys without enabling the **Token Exchange** grant type. Hence, you need to make sure the **Token Exchange** grant type is enabled when generating the keys. You can click on **Advanced Configurations** to check whether the **Token Exchange** grant type is enabled (By default this will be enabled). 
```

Comment:
I selected Token Exchange on generation credentials, but still don't see the User Keys tab. Can you please check these steps and verify with the latest UI in case there is a change. Once verified I can try out all the steps once again.

Reply:
Seems a fact is missing here. We need to have an already registered IdP(s) as well. What this tip says is correct. "The **User Keys** tab will be **disabled** if you .....". 
What is missing here is, in order for the **User Keys** tab to appear, you need to have an already registered IdP(s) in your organization. But I am not sure whether this is the place to say it or whether we should say it in the page **Registering an External Identity Provider (IdP) for Choreo API Management**

Reply:
Ack. I'll check and add that info in an appropriate manner.